src/thresholds.service.js

```diff
@@ -103,44 +28,51 @@ class ThresholdsService {
       return
     }
 
-    const { weeklyLevels, weeksOfStock, targetPopulations } = getFactors(stockCount, location)
+    const transformedLocation = transform(location, productCoefficients, stockCount.date)
```

Comment:
My original idea was to have use the location transformer as a normalization layer that can understand and unify different location schemas (therefore also the duplication of `level` mentioned below), but, the way it turned out at the end, I think your idea of just having it as a factor extractor is better.